src/conf.cpp

```diff
@@ -39,244 +39,138 @@ struct section {
     struct section *next;
 };
 
-ConfFile::~ConfFile()
+ConfFile::ConfFile(const char *filename)
+    : _filename(filename)
+    , _entry(nullptr)
+    , _last_section(nullptr)
+    , _n(0)
+    , _line(0)
```

Comment:
There is one big difference in the way I use configuration information in camera-straming-daemon and in mavlink-router. I won't know all the sections and keys expected in a configuration file at my main code. It is possible for users to create new classes and it would be nice to let them to access configuration files. And I plan to have configuration specific for gstreamer formats, so it will depend on formats supported in the board that the camera is running.
I don't agree that I am replacing the internal structure to store configurations with a map of strings. Settings class store in a string map the configuration merged for multiple files loaded with ConfFile. So I am using a map of strings instead of the mavlink-router "struct opt". The structures to store sections and configurations for a single file can be removed in both mavlink-router and camera-streaming-daemon.

In camera-streaming-daemon I'll need to access information from configuration files 
in different moments and not just in initialization. In mavlink-router the configuration is only used in initialization, so we won't need to keep structures holding this information. That's why I separated this in different places.
Another difference is that I don't know what 

1. I was planning to start using nmap in the future, so we won't need to allocate memory for lines of the file. So I agree.

2 and 3) nice idea and I really liked it. In this case we will only need to allocate memory for the value and it doesn't look hard to implement.
But I have one question. How can we handle multiple configuration files, with priorities, like it is handled in mavlink-router?

I agree that it is good to keep sync between mavlink-router and camera, that's why I asked for review in this.